pkg/table/tables/cache.go

```diff
@@ -261,6 +278,12 @@ func txnCtxAddCachedTable(sctx sessionctx.Context, tid int64, handle *cachedTabl
 
 // UpdateRecord implements table.Table
 func (c *cachedTable) UpdateRecord(ctx context.Context, sctx sessionctx.Context, h kv.Handle, oldData, newData []types.Datum, touched []bool) error {
+	txn, err := sctx.Txn(true)
+	if err != nil {
+		return err
+	}
+	defer handleTableSizeDelta(c, txn.Size(), txn)
```

Comment:
Why change the size when updating the record instead of removing the record?

Reply:
Because we just care the size limit of the table, we just need to check the growth of the table size, **remove will not make the table growth**.

One possible fail is that the user remove ... remove ... remove ... and insert/update, the table size is not accurate enough because we don't take the remove into account.
But the real table size is get from loading the cached, that's more accurate than calculate the size by transaction's size delta.